src/shims/unix/unnamed_socket.rs

```diff
@@ -347,13 +347,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let this = self.eval_context_mut();
 
         let pipefd = this.deref_pointer_as(pipefd, this.machine.layouts.i32)?;
-        let flags = match flags {
+        let mut flags = match flags {
             Some(flags) => this.read_scalar(flags)?.to_i32()?,
             None => 0,
         };
 
         // As usual we ignore CLOEXEC.
         let cloexec = this.eval_libc_i32("O_CLOEXEC");
+
+        // Interpret the flag. Every flag we recognize is "subtracted" from `flags`, so
+        // if there is anything left at the end, that's an unsupported flag.
+        let mut is_nonblock = false;
+        if flags & this.eval_libc_i32("O_NONBLOCK") == this.eval_libc_i32("O_NONBLOCK") {
+            is_nonblock = true;
+            flags &= !(this.eval_libc_i32("SOCK_NONBLOCK"));
```

Comment:
Why is this SOCK_NONBLOCK when the others are O_NONBLOCK?

Please call `eval_libc` only once here. Add a `let nonblock = this.eval_libc_i32("O_NONBLOCK")` just after `let cloexec`.